src/AlertBannerEntityAccessControlHandler.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+namespace Drupal\localgov_alert_banner;
+
+use Drupal\Core\Entity\EntityAccessControlHandler;
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\Session\AccountInterface;
+use Drupal\Core\Access\AccessResult;
+
+/**
+ * Access controller for the Alert banner entity.
+ *
+ * @see \Drupal\localgov_alert_banner\Entity\AlertBannerEntity.
+ */
+class AlertBannerEntityAccessControlHandler extends EntityAccessControlHandler {
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function checkAccess(EntityInterface $entity, $operation, AccountInterface $account) {
+    /** @var \Drupal\localgov_alert_banner\Entity\AlertBannerEntityInterface $entity */
+
+    switch ($operation) {
+
+      case 'view':
+
+        if (!$entity->isPublished()) {
+          $permission = $this->checkOwn($entity, 'view unpublished', $account);
```

Comment:
I guess this can become another issue, but will need tests for permissions. I wonder if they need to be so complex? Can they not just be a permission for who can access unpublished ones and everyone else who can't?

Reply:
I think permissions are going to be a seperarte issue. 
This is more an issue of what the scaffold generates, and there might be a need for the more granulated permissions anyway, so I'm not sure its something to worry about for alpha.
In terms of the workflow, at BHCC we allow all comms to write the banner, but only the emergency publisher role to publish. Though whether we should define that as a seperate role or a new permissions is something worth investigating. We also have the permission in #35.

As this blocks a lot of issues, i'd rather move this forward and open seperate issues to trim down what we don't need. We do need to write seperate tests to make sure only the right access is granted to author, edit, delete and publish, but first we need to discuss what thos should be.